### PR TITLE
ddclient: remove package and module on upstream maintainer request

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Description of changes

see https://github.com/ddclient/ddclient/issues/528

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/rename.nix [hunk 1]
```diff
@@ -54,7 +54,8 @@ in
     (mkRemovedOptionModule [ "services" "chronos" ] "The corresponding package was removed from nixpkgs.")
     (mkRemovedOptionModule [ "services" "couchpotato" ] "The corresponding package was removed from nixpkgs.")
     (mkRemovedOptionModule [ "services" "dd-agent" ] "dd-agent was removed from nixpkgs in favor of the newer datadog-agent.")
-    (mkRemovedOptionModule [ "services" "dnscrypt-proxy" ] "Use services.dnscrypt-proxy2 instead")
+    (mkRemovedOptionModule [ "services" "ddclient" ] "dd-agent was removed from nixpkgs in favor of the newer datadog-agent.")
```
Comment (RaitoBezarius) [inline nixos/modules/rename.nix:57]: you mixed up the renames

File: nixos/doc/manual/release-notes/rl-2311.section.md [hunk 2]
```diff
@@ -50,6 +50,8 @@
 
 - `util-linux` is now supported on Darwin and is no longer an alias to `unixtools`. Use the `unixtools.util-linux` package for access to the Apple variants of the utilities.
 
+- `services.ddclient` has been removed on the request of the upstream maintainer because it is unmaintained and has bugs. Please switch to a different software like inadyn.
```
Comment (RaitoBezarius) [inline nixos/doc/manual/release-notes/rl-2311.section.md:53]: Can you mention knsupdate too please?

**Comment by bjornfor** [comment]
> Please switch to a different software like `inadyn` or `knsupdate`.

Neither seem to have NixOS modules. Can we wait with removing `services.ddclient` until there's an easy migration path (NixOS module) please?

**Comment by SuperSandro2000** [comment]
The current release is broken e.g. https://github.com/ddclient/ddclient/issues/431

There will be no migration path from my side. I don't recommend people to run ddclient, it is probably not very secure and the latest release also not that reliable.

**Comment by bjornfor** [comment]
ddclient continues to be maintained by new maintainers: https://github.com/ddclient/ddclient/issues/528#issuecomment-1642163043. I guess we can revert this PR?

**Comment by SuperSandro2000** [comment]
There is no release yet, so the software itself has still all the known bugs.

**Pull request merged**